build/three.js

```diff
@@ -18336,17 +18329,32 @@
 
 				parameters.dithering ? '#define DITHERING' : '',
 
-				( parameters.outputEncoding || parameters.mapEncoding || parameters.matcapEncoding || parameters.envMapEncoding || parameters.emissiveMapEncoding || parameters.lightMapEncoding ) ?
-					ShaderChunk[ 'encodings_pars_fragment' ] : '', // this code is required here because it is used by the various encoding/decoding function defined below
+				ShaderChunk[ 'encodings_pars_fragment' ],
+
 				parameters.mapEncoding ? getTexelDecodingFunction( 'mapTexelToLinear', parameters.mapEncoding ) : '',
 				parameters.matcapEncoding ? getTexelDecodingFunction( 'matcapTexelToLinear', parameters.matcapEncoding ) : '',
 				parameters.envMapEncoding ? getTexelDecodingFunction( 'envMapTexelToLinear', parameters.envMapEncoding ) : '',
 				parameters.emissiveMapEncoding ? getTexelDecodingFunction( 'emissiveMapTexelToLinear', parameters.emissiveMapEncoding ) : '',
 				parameters.lightMapEncoding ? getTexelDecodingFunction( 'lightMapTexelToLinear', parameters.lightMapEncoding ) : '',
-				parameters.outputEncoding ? getTexelEncodingFunction( 'linearToOutputTexel', parameters.outputEncoding ) : '',
 
 				parameters.depthPacking ? '#define DEPTH_PACKING ' + parameters.depthPacking : '',
 
+				'uniform int outputEncoding;',
+
+				'vec4 linearToOutputTexel( vec4 value ) {',
+
+				'	if ( outputEncoding == 3000 ) return LinearToLinear( value );',
+				'	else if ( outputEncoding == 3001 ) return LinearTosRGB( value );',
+				'	else if ( outputEncoding == 3002 ) return LinearToRGBE( value );',
+				'	else if ( outputEncoding == 3003 ) return LinearToLogLuv( value );',
+				'	else if ( outputEncoding == 3004 ) return LinearToRGBM( value, 7.0 );',
+				'	else if ( outputEncoding == 3005 ) return LinearToRGBM( value, 16.0 );',
+				'	else if ( outputEncoding == 3006 ) return LinearToRGBD( value, 256.0 );',
+				'	else if ( outputEncoding == 3007 ) return LinearToGamma( value, float( GAMMA_FACTOR ) );',
+				'	else return LinearToLinear( value );',
+
+				'}',
+
```

Comment:
This should go in `encodings_pars_fragmen.glsl` I guess?

Reply:
Yep, that does work. However, `PMREMGenerator` needs some adjustments.

This class actually already used a similar approach like the renderer now. Meaning input and output encoding are passed in as uniforms. There might be some room for refactoring/clean up in `PMREMGenerator` if this PR gets merged.